### ISSUE TITLE
[FR][WORKFLOWS] Allow user to see jobs logs

### ISSUE THREAD

**Issue by rockandska** [ansible/awx#9033]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY

Not sure if it was implemented since i use an old version of AWX, but didn't find any related in awx issues and tower documentation.

I was expecting by give "execution" permission on a workflow to user, to let him only see the workflows he has access.
But unfortunately, event if the workflow seems to works, the user can't see the jobs logs of the workflow playbooks.
The only way i've found to let users see the logs, is to give "read" permission on ALL playbooks but this solution is not reliable in the time :
- the user see playbooks now and not  only the workflows
- it is very long to do when you have many workflows
- if new playbooks are added to the workflows, you need to update the permission regarding this new additions.

A perfect solution would be to propagate "read" permission only on logs concerned by the workflow.

Best regards,

**Comment by wenottingham** [comment]
I believe this is intentional - this is the only way you can delegate access to run a workflow without delegating some sort of view access on the individual templates. cc @AlanCoding ?

**Comment by AlanCoding** [comment]
Correct, these are managed separately. The UI should be savvy to the fact that users viewing a workflow may get 403s following links for templates or jobs therein.

As an enhancement, it's possible that we could add new roles for workflows that give roles to job templates or other things in it, but this would be a many-to-many mapping, which we have never done in any other scenario.

**Comment by NAOwens** [comment]
This would be a great feature to have.  I don't want to provide read access to the users that I'm allowing to run the workflow but I'd like them to see the log output so if it fails they can see why.  Having another permission field for templates to read logs would be perfect!
